js/plugins/ollama/src/index.ts

```diff
@@ -212,25 +230,51 @@ function toOllamaRequest(
     model: name,
     options,
     stream,
+    tools: input.tools?.filter(isValidOllamaTool).map(toOllamaTool),
   };
   if (type === 'chat') {
     const messages: Message[] = [];
     input.messages.forEach((m) => {
       let messageText = '';
+      const role = toOllamaRole(m.role);
       const images: string[] = [];
+      const toolRequests: ToolRequest[] = [];
+      const toolResponses: ToolResponse[] = [];
       m.content.forEach((c) => {
         if (c.text) {
           messageText += c.text;
         }
         if (c.media) {
           images.push(c.media.url);
         }
+        if (c.toolRequest) {
+          toolRequests.push(c.toolRequest);
+        }
+        if (c.toolResponse) {
+          toolResponses.push(c.toolResponse);
+        }
       });
-      messages.push({
-        role: toOllamaRole(m.role),
-        content: messageText,
-        images: images.length > 0 ? images : undefined,
-      });
+      if (role == 'tool') {
```

Comment:
I don't think you need to check for `role == 'tool'`, if message contains tool responses add them to the message, regardless of role.